### ISSUE TITLE
Quasar BEX v3 postMessage payload format

### ISSUE THREAD

**Issue by thomas66777** [quasarframework/quasar#9405]
I am trying to setup a simple window eventListener to go from webpage to extension. I realize that this logic for this has been abstracted inside the auto-generated file `bex/content/window-event-listener.js`

```js
export const listenForWindowEvents = (bridge, type) => {
  // Listen for any events from the web page and transmit to the BEX bridge.
  window.addEventListener('message', payload => {
...
    /**** payload.data is like an object right here ****/
    if (payload.data.from !== void 0 && payload.data.from === type) {

      /**** payload.data is like an array right here ****/
      const eventData = payload.data[0],
        bridgeEvents = bridge.getEvents()

      for (let event in bridgeEvents) {
        if (event === eventData.event) {
          bridgeEvents[event](eventData.payload)
        }
      }
    }
...
}
```

The problem is that `payload.data` is being used both like an Array and an Object.  It requires both `payload.data.from` and `payload.data[0]` to work.  I got it to work by setting up my webpage like this:
```js
var payload = {
    target: 'toExtension',
    from: 'bex-content-script',
    sender: 'localhost',
    0: {event:'bridge.myevent', payload:{x:'y'}},
}
window.postMessage(payload, "*");
```

I realize this is probably not the way I am supposed to setup my payload and was wondering if I am missing something or if `window-event-listener.js` is meant to have a different property for payload data.  Thank you

**Comment by rstoenescu** [comment]
Fix will be available through q/app-webpack v3.4.0

```
var payload = {
    target: 'toExtension',
    from: 'bex-content-script',
    sender: 'localhost',
    event:'bridge.myevent',
    payload: {x:'y'}
}

**Comment by rstoenescu** [comment]
I now realize that the initial code was correct. The `payload` is a MessageEvent and it is an Object with number keys and from key too. It wasn't a mistake. Reverting changes as it broke communication completely.

<img width="355" alt="image" src="https://user-images.githubusercontent.com/4665641/160609540-2f8f7f98-7bcc-4acd-80cd-770e0a2eae08.png">
